0004-post-tags.md

```diff
@@ -0,0 +1,335 @@
+Content for tags would be located in the respective root: `https://example.org/c/example/tag/tag`
+
+The ID of a CommunityPostTag is a URL prefixed with `https://instance/c/community/tags/` and a user-chosen or generated (from the name) ID. The ID itself will rarely be user-visible, since the name is used there.
+
+For now tags will only be applicable to posts, however the general design allows for them to be attached to any kind of object later on, be that instance, community or user. The limited initial scope allows for easier modifications should any rough edges or missing features be discovered.
+Additionally per Tag Inheritance could be added by expanding on this structure, more details can be found in [Future Possibilities > Tag Inheritance](#tag-inheritance).
+
+Example Tag Objects:
+
+```json
+{
+    "type": "lemmy:CommunityPostTag",
+    "url": "https://example.org/c/news/tag/newspaper-company-1",
+    "name": "Related to Newspaper Company 1"
+}
+```
+
+```json
+{
+    "type": "lemmy:CommunityPostTag",
+    "url": "https://example.org/c/news/tag/blood-gore",
+    "name": "Blood/Gore"
+}
+```
+
+```json
+{
+    "type": "lemmy:CommunityPostTag",
+    "url": "https://example.org/c/news/tag/news",
+    "name": "News"
+}
+```
+
+Below is an example of a News post in the news community. The post has a communtiy tag for the newspaper company that published it, as well as a content warning for Blood/Gore. In order to share this news post across instances it also is tagged with the news tag of a remote instance.
+
+Example Post json:
+
+``` json
+{
+  "@context": [
+    "https://www.w3.org/ns/activitystreams",
+    "https://join-lemmy.org/context.json"
+  ],
+  "to": [
+    "https://example.org/c/news",
+    "https://www.w3.org/ns/activitystreams#Public"
+  ],
+  "audience": "https://example.org/c/news",
+  "content": ". . .",
+  "sensitive": true,
+  "tag": [
+    {
+      "type": "lemmy:CommunityPostTag",
+      "url": "https://example.org/c/news/tag/newspaper-company-1",
+      "name": "Related to Newspaper Company 1"
+    },
+    {
+      "type": "lemmy:CommunityPostTag",
+      "url": "https://example.org/c/news/tag/blood-gore",
+      "name": "Blood/Gore"
+    },
+    {
+      "type": "lemmy:CommunityPostTag",
+      "url": "https://example.org/c/news/tag/news",
+      "name": "News"
+    }
+  ]
+}
+```
+
+### Backend:
+I am not quite familiar with the Lemmy Backend so any corrections or additions for this section are appreciated.
+
+One new table for "community_post_tag" would be required:
+
+```sql
+CREATE TABLE community_post_tag (
+  id bigserial PRIMARY KEY,
+  community_id bigint REFERENCES community.id,
```

Comment:
Were not using bigint/bigserial anywhere.

```suggestion
  id serial PRIMARY KEY,
  community_id int REFERENCES community.id,
```

Reply:
still no? it would make sense to start, for communities maybe 2B limit is fine, but for things like posts and tags probably no, esp. since tags can be arbitrary values created potentially (in the future at least) by anyone

Reply:
Even comments use normal int for id. And as tags can only be created by admins/mods, there will be much less of them than comments which can be created by anyone.